File: Interview_Prep_App/native_java/javatemplate/src/main/java/com/tc2r/javatemplate/repositories/models/Answer.java
package com.tc2r.javatemplate.repositories.models;

import android.os.Parcel;
import android.os.Parcelable;

import com.tc2r.javatemplate.repositories.models.response.DomainObject;

// FIXME: Create Response objects to handle data returned from api.
// Consider using JSON to POJO plugin or website convertor.

public class Answer implements DomainObject, Parcelable {


    private Integer id;
    private String answer;
    private String details;


    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public Answer() {
    }

    public Answer(String answer, String details) {
        this.answer = answer;
        this.details = details;
    }

    public String getAnswer() {
        return answer;
    }

    public void setAnswer(String answer) {
        this.answer = answer;
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }


    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.answer);
        dest.writeString(this.details);
    }

    protected Answer(Parcel in) {
        this.answer = in.readString();
        this.details = in.readString();
    }

    public static final Creator<Answer> CREATOR = new Creator<Answer>() {
        @Override
        public Answer createFromParcel(Parcel source) {
            return new Answer(source);
        }

        @Override
        public Answer[] newArray(int size) {
            return new Answer[size];
        }
    };
}

File: Interview_Prep_App/native_kotlin/retrofit2_moshi_Androidlion48/src/main/java/com/interviewprep/moshi_androidlion48/repository/models/Answer.kt
package com.interviewprep.moshi_androidlion48.repository.models

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

// FIXME: Create Response objects to handle data returned from api.
// Consider using JSON to POJO plugin or website convertor.
class Answer(
    var answer: String? = "",
    var details: String? = ""
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString(),
        parcel.readString()
    ) {
    }

    override fun describeContents(): Int {
        TODO("Not yet implemented")
    }

    override fun writeToParcel(p0: Parcel, p1: Int) {
        TODO("Not yet implemented")
    }

    companion object CREATOR : Creator<Answer> {
        override fun createFromParcel(parcel: Parcel): Answer {
            return Answer(parcel)
        }

        override fun newArray(size: Int): Array<Answer?> {
            return arrayOfNulls(size)
        }
    }
}
